### ISSUE TITLE
reson-relay-compiler not working for bs-platform@8.0.0+

### ISSUE THREAD

**Issue by Kasama** [zth/reason-relay#78]
`reason-relay-compiler` [references `bs-platform/lib/js/block.js`](https://github.com/zth/reason-relay/blob/d2ca7b81f9781eed64853f1ee30df5b048f6728e/packages/reason-relay/language-plugin/src/transformer/TypesTransformer.gen.tsx#L6). A file that was [deleted](https://github.com/BuckleScript/bucklescript/commit/bfcc48b337c3e7971966a0cae11887de7bda5bfe#diff-40231e0aa5ee32097fd3133156bbfe92) in version 8.0.0+ of `bs-platform`.

When trying to use the compiler with `yarn run reason-relay-compile`, it outputs the following error:

```
yarn run v1.17.3
$ reason-relay-compiler
Error: Cannot find module 'bs-platform/lib/js/block.js'
Require stack:
- /my-project/node_modules/reason-relay/language-plugin/reason-relay-language-plugin.js
- /my-project/node_modules/relay-compiler/bin/relay-compiler
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:1030:15)
    at Function.Module._load (internal/modules/cjs/loader.js:899:27)
    at Module.require (internal/modules/cjs/loader.js:1090:19)
    at require (internal/modules/cjs/helpers.js:75:18)
    at Object.<anonymous> (/my-project/node_modules/reason-relay/language-plugin/reason-relay-language-plugin.js:1:1023)
    at r (/my-project/node_modules/reason-relay/language-plugin/reason-relay-language-plugin.js:1:124)
    at Object.<anonymous> (/my-project/node_modules/reason-relay/language-plugin/reason-relay-language-plugin.js:1:1141723)
    at r (/my-project/node_modules/reason-relay/language-plugin/reason-relay-language-plugin.js:1:124)
    at Object.<anonymous> (/my-project/node_modules/reason-relay/language-plugin/reason-relay-language-plugin.js:1:1141087)
    at r (/my-project/node_modules/reason-relay/language-plugin/reason-relay-language-plugin.js:1:124)
```

Reproducing the issue
------------------------------

simplified files to reproduce the issue:

`package.json`
```json
{
  "name": "test",
  "version": "0.1.0",
  "scripts": {
    "relay": "reason-relay-compiler"
  },
  "author": "",
  "license": "MIT",
  "dependencies": {
    "graphql": "^14.6.0",
    "reason-relay": "^0.9.2",
    "relay-compiler": "10.0.0",
    "relay-config": "10.0.0",
    "relay-runtime": "10.0.0"
  },
  "devDependencies": {
    "bs-platform": "^8.0.3"
  }
}
```

`relay.config.js`
```js
// relay.config.js
module.exports = {
  src: ".",
  schema: "./schema.graphql",
  artifactDirectory: "./__generated__"
};
```

`schema.graphql` (it can be empty, included for completeness)
```graphql
schema {
  query: RootQueryType
}

interface Node {
  "The id of the object."
  id: ID!
}

type RootQueryType {
  node(
    "The id of an object."
    id: ID!
  ): Node
}
```

with all those files in a directory, run
```bash
$ yarn install
$ yarn run reason-relay-compiler
```

**Comment by zth** [comment]
Hi!

I haven't gotten around to upgrading `bs-platform` to 8.x.x just yet (I should clarify in the docs that 7.x.x is the only thing that'll work right now). For now, please remain at `7.3.2` and Relay version `9.0.0`. I'll revisit this soon again!

**Comment by osener** [comment]
The generated code seems to work with 8.x.x however. Can reason-relay somehow depend on this specific version and have a duplicate `bs-platform` installation under a package-level `node_modules` folder?

**Comment by zth** [comment]
@osener yeah, that's probably the best way to approach it. I'll start working on this soon, hopefully it'll be an easy fix, but we'll see 😉

**Comment by zth** [comment]
Unfortunately a new release of ReasonRelay will have to wait some, but at least it's fixed now.

**Comment by zth** [comment]
Now released in `0.10.0`!

**Comment by osener** [comment]
Awesome release, thank you!
